decoder/semantic_tokens.go

```diff
@@ -218,6 +218,28 @@ func (d *Decoder) tokensForExpression(expr hclsyntax.Expression, constraints Exp
 
 			}
 		}
+		_, ok = constraints.TypeDeclarationExpr()
+		if ok {
+			tokens = append(tokens, lang.SemanticToken{
+				Type:      lang.TokenTypePrimitive,
+				Modifiers: []lang.SemanticTokenModifier{},
+				Range:     expr.Range(),
+			})
+		}
+	case *hclsyntax.FunctionCallExpr:
+		_, ok := constraints.TypeDeclarationExpr()
+		if ok {
+			tokens = append(tokens, lang.SemanticToken{
+				Type:      lang.TokenTypeCapsule,
+				Modifiers: []lang.SemanticTokenModifier{},
+				Range:     expr.Range(),
```

Comment:
```suggestion
				Range:     eType.NameRange,
```
Here I think that on the client/editor side we probably wouldn't want to highlight brackets with the same colour as the type name itself. Usually brackets, dots, commas etc. are not reported as highlightable tokens and they're already covered by the static grammar on client side.

Also the LSP isn't super clear about overlapping tokens and VS Code AFAIK doesn't handle those well.